packages/core/android/src/test/kotlin/com/datadog/reactnative/DdSdkBridgeExtTest.kt

```diff
@@ -0,0 +1,382 @@
+        testArray(nestedArray)
+    }
+
+    @Test
+    fun `M do a proper conversion and not throw W toMap { malformed nested map }`() {
+        // Given
+        val readableMap =
+            mapOf(
+                "map" to mapOf("test" to "test") // NOT a ReadableMap
+            ).toReadableMap()
+
+        // When
+        val map = readableMap.toMap()
+
+        // Then
+        assertThat(map).isEmpty()
+    }
+
+    @Test
+    fun `M do a proper conversion and not throw W toList { malformed nested array }`() {
+        // Given
+        val readableArray =
+            listOf(
+                mapOf("test" to "test"), // NOT a ReadableMap
+                listOf(null, true, 1.23, "test") // NOT a ReadableArray
+            ).toReadableArray()
+
+        // When
+        val list = readableArray.toList()
+
+        // Then
+        assertThat(list).isEmpty()
+    }
+
+    private fun getTestMap(): MutableMap<String, Any?> = mutableMapOf(
+        "null" to null,
+        "int" to 1,
+        "long" to 2L,
+        "float" to 3.0f,
+        "double" to 4.0,
+        "string" to "test",
+        "boolean" to true
+    )
+
+    private fun getTestWritableMap(): JavaOnlyMap = JavaOnlyMap.of(
+        "null",
+        null,
+        "int",
+        1,
+        "long",
+        2L,
+        "float",
+        3.0f,
+        "double",
+        4.0,
+        "string",
+        "test",
+        "boolean",
+        true
+    )
+
+    // Long type is not handled in JavaOnlyArray getType() in RN 0.71.10
+    // https://github.com/facebook/react-native/pull/43158
+    private fun getTestArray(): MutableList<Any?> = mutableListOf(
+        null,
+        1,
+        2.0f,
+        3.0,
+        "test",
+        true
+    )
+
+    // Long type is not handled in JavaOnlyArray getType() in RN 0.71.10
+    // https://github.com/facebook/react-native/pull/43158
+    private fun getTestWritableArray(): WritableArray = JavaOnlyArray.from(
+        listOf(
+            null,
+            1,
+            2.0f,
+            3.0,
+            "test",
+            true
+        )
+    )
+
+    private fun testWritableArray(array: WritableArray) {
+        assertThat(array.getDynamic(0).isNull).isEqualTo(true)
+
+        assertThat(array.getInt(1)).isInstanceOf(java.lang.Integer::class.java)
+        assertThat(array.getInt(1)).isEqualTo(1)
+
+        assertThat(array.getDouble(2)).isInstanceOf(java.lang.Double::class.java)
+        assertThat(array.getDouble(2)).isEqualTo(2.0)
+
+        assertThat(array.getDouble(3)).isInstanceOf(java.lang.Double::class.java)
+        assertThat(array.getDouble(3)).isEqualTo(3.0)
+
+        assertThat(array.getString(4)).isInstanceOf(java.lang.String::class.java)
+        assertThat(array.getString(4)).isEqualTo("test")
+
+        assertThat(array.getBoolean(5)).isInstanceOf(java.lang.Boolean::class.java)
+        assertThat(array.getBoolean(5)).isEqualTo(true)
```

Comment:
```suggestion
        assertThat(array.getBoolean(5)).isTrue
```

same changes apply below